### DISCUSSION TITLE
How to download pdf from server?

### DISCUSSION THREAD

**Discussion by melokki** [nuxt/nuxt#20154] [Questions]
Hello everyone

I am trying to download a pdf file from my api, but I am not able to succeed in any way I've tried so far.

My api return a stream and I want to download that in my nuxt project.
This is the code that I have right now

```ts

// server/api/invoices/show.get.ts
 return await $fetch(`/invoices/1234`, {
    baseURL,
    headers: {
      Authorization: `Bearer ${token}`,
    },
  });

```

and in my component I've tried to download the file like this

```ts
const response = await $fetch("/api/invoices/show");

  const blob = new Blob([response], {
    type: "application/pdf",
  });
  window.open(URL.createObjectURL(blob));

```

this code downloads a pdf file into my pc, but the pdf is empty, it's just a plain white page

the `response` looks something like this

```
%PDF-1.4\n1 0 obj\n<<\n/Title (��\u0000F\u0000a\u0000c\u0000t\u0000u\u0000r\u0000a)\n/Creator (��\u0000w\u0000k\u0000h\u0000t\u0000m\u0000l\u0000t\u0000o\u0000p\u0000d\u0000f\u0000 \u00000\u0000.\u00001\u00002\u0000.\u00006\u0000.\u00001)\n/Producer (��\u0000Q\u0000t\u0000 \u00004\u0000.\u00008\u0000.\u00007)\n/CreationDate (D:20230407175922+03'00'
```

I ran out of ideas and don't know what to do from here

Does someone has an idea about how to make this work?

**Comment by melokki** [comment]
found the following error while opening the pdf in browser

```
Warning: Invalid stream: "FormatError: Bad FCHECK in flate stream: 120, 239"
```

really don't know what to do from here

my api returns a stream and if I'm accessing the api url directly, I get the desired pdf, but if I make the request from nuxt `server/api/invoices/show.get.ts` I am getting the above error.

**Comment by BracketJohn** [comment]
You can do it as follows:
```ts
  const { data } = await useFetch('/api/pdf', { method: 'post', body: creditReportIds, responseType: 'blob' })
  const file = window.URL.createObjectURL(data);
  window.location.assign(file);
```

`ofetch` aka. `usefetch` supports the `responseType` attribute to get a blob. Read more: https://github.com/unjs/ofetch#%EF%B8%8F-parsing-response
